theseus/optimizer/sparse_linearization.py

```diff
@@ -152,12 +158,33 @@ def _atb_impl(self) -> torch.Tensor:
             A_col_ind = A_row_ptr.new_tensor(self.A_col_ind)
 
             # unsqueeze at the end for consistency with DenseLinearization
-            self._Atb = tmat_vec(
-                self.objective.batch_size,
+            self._Atb = sparse_mtv(
                 self.num_cols,
                 A_row_ptr,
                 A_col_ind,
-                self.A_val,
-                self.b,
+                self.A_val.double(),
```

Comment:
So our sparse solvers only supports `float64`?

Reply:
That's correct. This part of the code is also already merged.